### ISSUE TITLE
Farmer stuck in loop?

### ISSUE THREAD

**Issue by OCDCleric** [ldtteam/minecolonies#3658]
### Minecolonies version
<0.10.293>

Version: 1.12.2

### Expected behavior

- 
Farmer is moving back and forth from the Farm block to the Field block and returns to Farm block repeatedly. Only exception is when it sleeps at house.

### Actual behavior
I used the commands to observe current activity. It goes as follows.
1.Preparing
2.FARMER_PLANT
3.Idle
4.Inventory_full

Number 2 doesn't function as the moment it reaches field it moves to Number 3.
As for Number 4 this shouldn't happen as it uses only 3 inventory slots for items currently.

### Steps to reproduce the problem

1. Build acacia farm.
2. Build Acacia Plain Field
3. Supply farmer with tool/seed
4. ... Profit

**Comment by Asherslab** [comment]
Have you put seeds inside the field block?

**Comment by OCDCleric** [comment]
Yes i have.

**Comment by Asherslab** [comment]
did you use modded, or vanilla seeds? if modded, which mod?

**Comment by OCDCleric** [comment]
Vanilla seeds and potatoes gonna check if this is a mod error due to Pams Harvestcraft.

**Comment by Asherslab** [comment]
the mod will only have issues with pams if you're using pams seeds themselves

**Comment by Raycoms** [comment]
Does this happen after restart?

If so, can you send us the world?

**Comment by OCDCleric** [comment]
Seems so. The farmer still stuck in loop.

**Comment by OCDCleric** [comment]
Ill try a new world before sending old one.

**Comment by Raycoms** [comment]
If you can reproduce it in a new world, even better

**Comment by OCDCleric** [comment]
Reproduced. I quick started a superflat and straight away built farmer and field. Same loop occurs. May take a bit to get the world uploaded. Only internet here is my cell phone with passable speeds lol.

**Comment by OCDCleric** [comment]
[New World.zip](https://github.com/ldtteam/minecolonies/files/3263950/New.World.zip)
Made a .zip file of world since sending via phone...

**Comment by fr3akb0y** [comment]
My Farmer also got stuck on the Acacia Farm from the schematics. It was because of the half slab above the water. Removing these fixed the Farmer again (for me).

**Comment by OCDCleric** [comment]
Confirmed. Acacia slabs on 'Acacia Plain Field' was issue. Wonder why... Im gonna look at other field types in schematic.

Edit: Seems acacia is only field that surrounds Field block with other blocks. Maybe that interferes with npc"s interacting with the block? Gonna test a few others just to be sure.

Edit 2: Ok I did a few more builds. Turns out if you surround the field block with water in the ground and slabs above it the farmer goes into the loop mentioned. Don't know why but that water/slab combo is the cause and just removing the slabs is the fix. Weird...
